src/components/Chart/data/bitDataAccumulator.js

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2015 - 2018, Nordic Semiconductor ASA
+ *
+ * All rights reserved.
+ *
+ * Use in source and binary forms, redistribution in binary form only, with
+ * or without modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions in binary form, except as embedded into a Nordic
+ *    Semiconductor ASA integrated circuit in a product or a software update for
+ *    such product, must reproduce the above copyright notice, this list of
+ *    conditions and the following disclaimer in the documentation and/or other
+ *    materials provided with the distribution.
+ *
+ * 2. Neither the name of Nordic Semiconductor ASA nor the names of its
+ *    contributors may be used to endorse or promote products derived from this
+ *    software without specific prior written permission.
+ *
+ * 3. This software, with or without modification, must only be used with a Nordic
+ *    Semiconductor ASA integrated circuit.
+ *
+ * 4. Any software provided in binary form under this license must not be reverse
+ *    engineered, decompiled, modified and/or disassembled.
+ *
+ * THIS SOFTWARE IS PROVIDED BY NORDIC SEMICONDUCTOR ASA "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY, NONINFRINGEMENT, AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL NORDIC SEMICONDUCTOR ASA OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+/* eslint no-plusplus: off */
+
+import bitDataStorage from './bitDataStorage';
+import { nbDigitalChannels, options } from '../../../globals';
+import {
+    always0,
+    always1,
+    averagedBitState,
+    sometimes0And1,
+} from '../../../utils/bitConversion';
+
+export default () => ({
```

Comment:
Just a question really, but why not use classes for bitsDataAccumulator, bitsDataSelector, bitsDataStorage and dataAccumulator? Not that they currently provide much more than not having to create your own constructor at the moment, but you do get the possibility of inheritance for example if that would be relevant.

Reply:
It is very seldom that I see advantages in using classes. And since I prefer composition over inheritance in most cases, I see the possibility of easily getting inheritance more as a danger than a benefit. :-) 

One thing where classes can be an advantage is when I want to make it explicit that [fields](https://github.com/tc39/proposal-class-fields) or [methods](https://github.com/tc39/proposal-private-methods) are private. But in most code bases I do not think that advantage is big enough to actually start using classes there.